samples/apps/forum/src/controllers/site.ts

```diff
@@ -480,6 +556,7 @@ ${HEADER_HTML}
 
 <script>
 async function main() {
+    await initUser()
```

Comment:
There seems to be a race here with the call that's made on DOMContentLoaded, and I lose it about half the time. At which point the page sends a request with an empty jwt and gets a rather unfriendly error from the server.

This isn't an issue on the form pages because of the human-induced delay.

awaiting initUser at the beginning of each main() is tempting, but I suspect this is not what a sophisticated web developer would do, so I defer to @letmaik :)

Reply:
You are referring to the statistics page I assume. Yeah, it's all a bit hacky... definitely a race there. Your suggestion is fine. In a proper project you'd use some framework that deals with all of this.